internal/service/cognitoidp/user.go

```diff
@@ -0,0 +1,535 @@
+			Username:   aws.String(d.Get("username").(string)),
+			UserPoolId: aws.String(d.Get("user_pool_id").(string)),
+		}
+
+		_, err := conn.AdminDisableUser(disableParams)
+		if err != nil {
+			return fmt.Errorf("Error disabling Cognito User (%s): %w", d.Id(), err)
+		}
+	}
+
+	if v, ok := d.GetOk("password"); ok {
+		setPasswordParams := &cognitoidentityprovider.AdminSetUserPasswordInput{
+			Username:   aws.String(d.Get("username").(string)),
+			UserPoolId: aws.String(d.Get("user_pool_id").(string)),
+			Password:   aws.String(v.(string)),
+			Permanent:  aws.Bool(true),
+		}
+
+		_, err := conn.AdminSetUserPassword(setPasswordParams)
+		if err != nil {
+			return fmt.Errorf("Error setting Cognito User's password (%s): %w", d.Id(), err)
+		}
+	}
+
+	return resourceUserRead(d, meta)
+}
+
+func resourceUserRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).CognitoIDPConn
+
+	log.Println("[DEBUG] Reading Cognito User")
+
+	params := &cognitoidentityprovider.AdminGetUserInput{
+		Username:   aws.String(d.Get("username").(string)),
+		UserPoolId: aws.String(d.Get("user_pool_id").(string)),
+	}
+
+	user, err := conn.AdminGetUser(params)
+	if err != nil {
+		if tfawserr.ErrCodeEquals(err, cognitoidentityprovider.ErrCodeUserNotFoundException) {
+			log.Printf("[WARN] Cognito User %s not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Cognito User (%s): %w", d.Id(), err)
+	}
+
+	if err := d.Set("attributes", flattenUserAttributes(user.UserAttributes)); err != nil {
+		return fmt.Errorf("failed setting user attributes (%s): %w", d.Id(), err)
+	}
+
+	if err := d.Set("mfa_setting_list", user.UserMFASettingList); err != nil {
+		return fmt.Errorf("failed setting user's mfa settings (%s): %w", d.Id(), err)
+	}
+
+	d.Set("preferred_mfa_setting", aws.StringValue(user.PreferredMfaSetting))
+	d.Set("status", aws.StringValue(user.UserStatus))
+	d.Set("enabled", aws.BoolValue(user.Enabled))
+	d.Set("creation_date", user.UserCreateDate.Format(time.RFC3339))
+	d.Set("last_modified_date", user.UserLastModifiedDate.Format(time.RFC3339))
+	d.Set("sub", retrieveUserSub(user.UserAttributes))
+
+	return nil
+}
+
+func resourceUserUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).CognitoIDPConn
+
+	log.Println("[DEBUG] Updating Cognito User")
+
+	if d.HasChange("attributes") {
+		old, new := d.GetChange("attributes")
+
+		upd, del := computeUserAttributesUpdate(old, new)
+
+		if len(upd) > 0 {
+			params := &cognitoidentityprovider.AdminUpdateUserAttributesInput{
+				Username:       aws.String(d.Get("username").(string)),
+				UserPoolId:     aws.String(d.Get("user_pool_id").(string)),
+				UserAttributes: expandAttribute(upd),
+			}
+
+			if v, ok := d.GetOk("client_metadata"); ok {
+				metadata := v.(map[string]interface{})
+				params.ClientMetadata = expandUserClientMetadata(metadata)
```

Comment:
When I was testing, I noticed that `AdminUpdateUserAttributesInput` also has a `ClientMetadata` field. So I believe it should be added here as well.